formio-widget/src/index.js

```diff
@@ -2,112 +2,151 @@ window.onload = function () {
-    document.getElementById("formio"),
-    config.baseUrl +
-      "/" +
-      form_metadata.project_name +
-      "/" +
-      form_metadata.form_name,
-    {
-      //Turn off default buttons
-      buttonSettings: {
-        showCancel: false,
-        showPrevious: false,
-        showNext: false,
-        showSubmit: false,
-      },
-      i18n: {
-        en: { pattern: "Must use the format shown" },
-      },
-    }
-  ).then(function (wizard) {
-    window.form = wizard;
-    formName = form._form.title;
-    formModified = form._form.modified;
-
-    //Force new tab on formlinks
-    config.body_container.on(
-      "click",
-      config.formio_container + " a",
-      function (e) {
-        e.target.target = "_blank";
-      }
-    );
-
-    //Change event/GTM
-    wizard.on("click", function (change) {
-      var changeObj = change;
-      if (
-        typeof changeObj.changed != "undefined" &&
-        typeof changeObj.changed.component != "undefined"
-      ) {
-        dataLayer.push({
-          event: "formio-interaction",
-          "formio-name": formName,
-          "formio-input-id": changeObj.changed.component.id,
-          "formio-input-type": changeObj.changed.component.type,
-          "formio-input-value": changeObj.changed.component.value,
-          "formio-input-key": changeObj.changed.component.key,
-          "formio-input-label-raw": changeObj.changed.component.label,
-          "formio-version": formModified,
-          "formio-category": "Form: " + formName,
-          "formio-action": "Value changed",
-        });
-      }
-    });
+    
+  // polyfill plugin function to fix the namespace option doesn't pass to Formio.makeRequest
+  var NamespacePolyfillPlugin = {
+    priority: 0,
+    preRequest: function(requestArgs) {
+        if (requestArgs.formio && formioInstances) {
+            const formioInstance = formioInstances.find((item)=>item.formio.formUrl===requestArgs.formio.formUrl);
+            requestArgs.formio = formioInstance.formio;
+            requestArgs.opts.formio = formioInstance.formio;
+        }
+        if (requestArgs.formio.options) requestArgs.opts.namespace = requestArgs.formio.options.namespace
+      return Promise.resolve(null);
+    },
+  }  
+  Formio.registerPlugin(NamespacePolyfillPlugin, 'namespacepolyfill');
 
-    //Must use 'applicationSubmit' custom event on primary submit
-    wizard.on("applicationSubmit", function (data) {
-      config.submitBtn.attr("disabled", true);
-      wizard
-        .submit()
-        .then(function () {
-          if (form_metadata.form_confirmation) {
-            window.location.href = "/" + form_metadata.form_confirmation;
+  if (formioInstances && formioInstances.length) {
+    // loop through all formioInstances and render the form for each instance
+      formioInstances.forEach(function(formioInstance) {
+          /*
+          * setup config
+          */
+          const baseUrl = "https://" + formioInstance.env_url.trim();
```

Comment:
How we can define a constant inside a loop? Either it can't be overwritten, and doing it multiple times is pointless (so it should be outside the loop), or it can be overwritten, in which case it's not really a constant.

Reply:
const in JS is block-scoped to the loop function so it won't be overwritten
the reason i didn't use `let` or `var` outside the loop is mainly for readability and has a better blcok structure
performance wise the differnce is minimise